src/pgduckdb_types.cpp

```diff
@@ -901,13 +902,19 @@ HeapTupleFetchNextColumnDatum(TupleDesc tupleDesc, HeapTuple tuple, HeapTupleRea
                               int att_num, bool *is_null) {
 	HeapTupleHeader tup = tuple->t_data;
 	bool hasnulls = HeapTupleHasNulls(tuple);
+	bool natts = HeapTupleHeaderGetNatts(tup);
 	int attnum;
 	char *tp;
 	uint32 off;
 	bits8 *bp = tup->t_bits;
 	bool slow = false;
 	Datum value = (Datum)0;
 
+	if (natts + 1 < att_num) {
```

Comment:
I think this shouldn't have the `+ 1`, but removing that breaks other tests even more...
```suggestion
	if (natts < att_num) {
```